apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1308 @@
+				if (request->entity == NULL) {
+					PRNT("no memory allocated");
+					return -1;
+				}
+				g_is_test_entity = 1;
+				memset(request->entity, '1', t);
+				request->entity[t] = '\0';
+			} else {
+				PRNT("test_entity is too big");
+				return -1;
+			}
+		}
+		else  if (strncmp(p, "keep-alive", 10) == 0) {
+			g_keep_alive_count = atoi(q)-1;
+		}
+		else if (strncmp(p, "chunk-count", 11) == 0) {
+			request->encoding = 1;
+			g_chunk_count = atoi(q);
+		}
+		else if (strncmp(p, "trailer", 7) == 0) {
+			g_trailer = atoi(q);
+		}
+		else {
+			PRNT("issue with optional args");
+			return -1;
+		}
+	}
+
+	request->buflen = WS_TEST_BUF_SIZE;
+	FUNC_EX;
+	return 0;
+}
+
+static int ws_test_process_query(int argc, char** argv)
+{
+	FUNC_EN;
+	struct http_client_request_t request;
+	struct http_keyvalue_list_t headers;
+	struct http_client_response_t response;
+	struct http_client_ssl_config_t *ssl_config = NULL;
+
+	int result = ws_test_prepare_request(argc, argv, &request);
+	if (result != 0) {
+		ws_test_dump_usage();
+		if (g_is_test_entity) {
+			free(request.entity);
+		}
+		return result;
+	}
+	ssl_config = g_https ? &g_ssl_config : NULL;
+
+	// before sending request, initialize keyvalue list for request headers
+	result = http_keyvalue_list_init(&headers);
+	if (result != 0) {
+		PRNT("http_keyvalue_list_init() failed %d", result);
+		goto release_out;
+	}
+
+	//keep alive header
+	if (g_keep_alive_count > 0) {
+		http_keyvalue_list_add(&headers, HEADERFIELD_CONNECT, HEADERFIELD_KEEPALIVE);
+	} else {
+		http_keyvalue_list_add(&headers, HEADERFIELD_CONNECT, HEADERFIELD_CLOSE);
+	}
+	http_keyvalue_list_add(&headers, HEADERFIELD_USERAGENT, HEADERFIELD_TINYARA);
+	request.headers = &headers;
+
+	// before sending request by sync function, must initialize response
+	if (ws_test_http_client_response_init(&response) < 0) {
+		PRNT("fail to init response");
+		goto release_out;
+	}
+	if (ws_test_http_client_send_request(&request, ssl_config, &response, NULL)) {
+		PRNT("fail to send request");
+		goto release_out;
+	}
+	if (response.status == 200) {
+		g_tc_pass++;
+	}
+	printf("----------sync response----------\n");
+	printf("status: %d %s\n", response.status, response.phrase);
+	printf("entity_len : %d total len : %d\n", response.entity_len, response.total_len);
+	printf("%s\n", response.entity);
+	printf("----====----\n");
+	printf("%s\n", response.message);
+	printf("--------- response ended --------\n");
+
+release_out:
+	if (g_is_test_entity) {
+		free(request.entity);
+	}
+	PRNT("going for: http_keyvalue_list_release(&headers)");
+	result = http_keyvalue_list_release(&headers);
+	if (result != 0) {
+		PRNT("http_keyvalue_list_release() failed %d", result);
+	}
+
+	ws_test_http_client_response_release(&response);
+	PRNT("request ended");
+
+	FUNC_EX;
+	return 0;
+}
+
+// webserver_test iters req_type https://127.0.0.1 chunk trailer keep-alive,
+int webserver_test_main(int argc, char *argv[])
+{
+	FUNC_EN;
+	int result = -1;
```

Comment:
at some places #define WGET_OK, WGET_ERR is used, while at other places -1, 0 is used.
its better to maintain a consistency